Lib/pydoc.py

```diff
@@ -1756,6 +1756,14 @@ def doc(thing, title='Python Library Documentation: %s', forceload=0,
     if output is None:
         try:
             what = thing if isinstance(thing, str) else type(thing).__name__
```

Comment:
This line is now redundant, since `what` is now always overridden in one of the two branches below:

```suggestion
```